app/services/calculators/conviction_end_date_calculator.rb

```diff
@@ -0,0 +1,7 @@
+module Calculators
+  class ConvictionEndDateCalculator < BaseCalculator
+    def expiry_date
+      disclosure_check.known_date + disclosure_check.conviction_length.send(disclosure_check.conviction_length_type.singularize)
```

Comment:
We can do the following and that way avoid the `#send` method that always give me the creeps 😄 

```ruby
disclosure_check.known_date.advance(disclosure_check.conviction_length_type.to_sym => disclosure_check.conviction_length)
```

Ideally as we will be doing this operation many times, we can create in the superclass a method `#conviction_length` that returns the hash already: 

```ruby
def conviction_length
  { disclosure_check.conviction_length_type.to_sym => disclosure_check.conviction_length }
end
```

So we can just do this:

```ruby
disclosure_check.known_date.advance(conviction_length)
```